packages/fx-core/src/plugins/resource/aad/plugin.ts

```diff
@@ -321,6 +326,34 @@ export class AadAppForTeamsImpl {
     return ResultFactory.Success(result);
   }
 
+  public async checkPermission(ctx: PluginContext): Promise<AadResult> {
+    TelemetryUtils.init(ctx);
+    Utils.addLogAndTelemetry(ctx.logProvider, Messages.StartCheckPermission);
+
+    await TokenProvider.init(ctx, TokenAudience.Graph);
+    const config = new CheckPermissionConfig();
+    await config.restoreConfigFromContext(ctx);
+
+    const userObjectId = config?.userInfo["aadId"];
+    const isAadOwner = await AadAppClient.checkPermission(
+      ctx,
+      Messages.EndCheckPermission.telemetry,
+      config.objectId!,
+      userObjectId
+    );
+
+    const result = [
+      {
+        name: Constants.permissions.name,
+        type: Constants.permissions.type,
+        roles: isAadOwner ? [Constants.permissions.owner] : [Constants.permissions.noPermission],
+        resourceId: config.objectId!,
+      },
+    ];
+    Utils.addLogAndTelemetry(ctx.logProvider, Messages.EndCheckPermission);
```

Comment:
where we send the error telemetry?

Reply:
Error telemetry will be sent in index.ts. 
For other info that needs to add to telemetry, will update necessary info in telemetry in another pr after we have confirmed required info with yanliang.